src/main.ml

```diff
@@ -115,9 +123,8 @@ let handle_connection xen_api_uri fd =
       )
   in
 
-  let channel = Nbd_lwt_unix.of_fd fd in
-  Lwt.finalize
-    (fun () ->
+  Nbd_lwt_unix.with_channel fd tls_role
+    (fun channel ->
        Nbd_lwt_unix.Server.connect channel ()
```

Comment:
Could we use `Server.with_connect` here? (First it needs to pass the export name to its argument function.)